arch/ARC/ARCDisassembler.c

```diff
@@ -200,7 +200,9 @@ static DecodeStatus DecodeMEMrs9(MCInst *Inst, unsigned Insn, uint64_t Address,
 	// We have the 9-bit immediate in the low bits, 6-bit register in high bits.
 	unsigned S9 = Insn & 0x1ff;
 	unsigned R = (Insn & (0x7fff & ~0x1ff)) >> 9;
-	DecodeGPR32RegisterClass(Inst, R, Address, Decoder);
+	if (!DecodeGPR32RegisterClass(Inst, R, Address, Decoder)) {
```

Comment:
```suggestion
	if (DecodeGPR32RegisterClass(Inst, R, Address, Decoder) == MCDisassembler_Fail) {
```

Please do the check explicitly. We never know if they might decide changing the values.